code/common/checkinputs.q

```diff
@@ -0,0 +1,257 @@
+    input:dict parameter;
+    if[not 11h~abs type key input;
+        '`$.schema.errors[`filterkey;`errormessage],.schema.examples[`filters1;`example]];
+    (input`testnotfunc):enlist(not;in;10 30);
+    filterpairs:raze value input;
+    if[any not in[count each filterpairs;2 3];
+        '`$.schema.errors[`filterpair;`errormessage],.schema.examples[`filters1;`example]];
+    if[not 15 in value each first each filterpairs where 3=count each filterpairs;
+        '`$.schema.errors[`filternot;`errormessage],.schema.examples[`filters1;`example]];
+    nots:where(~:)~/:ops:first each filterpairs;
+    notfilters:@\:[;1]filterpairs nots;
+    if[not all in[ops;.schema.allowedops];'`$.schema.errors[`allowedops;`errormessage]];
+    if[not all in[notfilters;.schema.allowednot];'`$.schema.errors[`allowednot;`errormessage]];
+    .checkinputs.withincheck'[filterpairs];
+    .checkinputs.inequalitycheck'[filterpairs];
+    :dict;
+  };
+
+withincheck:{[pair]
+    if[(("within"~string first pair)| "within[~:]"~string first pair)& 2<>count last pair;
+        '`$.schema.errors[`withincheck;`errormessage],.schema.examples[`filters3;`example]];};
+
+inequalitycheck:{[pair]
+    errmess:.schema.errors[`inequalities;`errormessage],.schema.examples[`filters3;`example];
+    errmess2:.schema.errors[`equalities;`errormessage],.schema.examples[`filters3;`example];
+    if[(("~<"~string first pair)|(enlist"<")~string first pair)& 1<>count last pair;
+        '`$errmess2];
+    if[(("~>"~string first pair)|(enlist">")~string first pair)& 1<>count last pair;
+        '`$errmess2];
+    if[(((enlist"=")~string first pair)|(enlist"~")~string first pair)&1<>count last pair;
+        '`$errmess2];
+    if[("~="~string first pair)&1<>count last pair;
+        '`$errmess];};
+
+// check that ordering parameter contains only symbols and is paired in the format
+// (direction;column).
+checkordering:{[dict;parameter]
+    if[11h=type (dict parameter);(dict parameter):enlist (dict parameter)];
+    input:dict parameter;
+    if[11h<>type raze input;
+        '`$.schema.errors[`checkorderingpair;`errormessage],.schema.examples[`ordering1;`example]];
+    if[0<>count except[count each input;2];
+        '`$.schema.errors[`checkorderingarrangment;`errormessage],.schema.examples[`ordering1;`example]];
+    if[0<>count except[first each input;`asc`desc];
+        '`$.schema.errors[`checkorderingdirection;`errormessage],.schema.examples[`ordering1;`example]];
+    $[in[`grouping;key dict];grouping:(dict`grouping),();grouping:()];
+    $[in[`timebar;key dict];timebar:(dict`timebar);timebar:()];
+    if[in[`aggregations;key dict];
+        aggs:dict`aggregations;
+        aggs:flip(key[aggs]where count each get aggs;raze aggs);
+        names:{
+        if[1=count x[1];
+            :`$(string x[0]),@[string x[1];0;upper]];
+        if[2=count x[1];
+            :`$(string x[0]),(@[string first x[1];0;upper]),@[string last x[1];0;upper]]
+        }'[aggs];
```

Comment:
Indentation needs fixed and I think this works out the same. If the count is always 1 or 2 can drop the if condition.
```suggestion
        names:{
            if[count[x 1]in 1 2;
                :`$raze string[x 0],.[string (),x 1;(::;0);upper]
        }'[aggs];
```